iis-workflows/iis-workflows-citationmatching/src/main/resources/eu/dnetlib/iis/workflows/citationmatching/chain/oozie_app/workflow.xml

```diff
@@ -52,319 +52,55 @@
-                </property>
-                <property>
-                    <name>workingDirectory</name>
-                    <value>${nameNode}${workingDir}/coansys</value>
-                </property>
-                <property>
-                    <name>sourceEntities</name>
-                    <value>${nameNode}${workingDir}/source_entities</value>
-                </property>
-                <property>
-                    <name>destinationEntities</name>
-                    <value>${nameNode}${workingDir}/destination_entities</value>
-                </property>
-                <property>
-                    <name>output</name>
-                    <!-- TODO MiconCodeReview: In case of changing action name, this should become '${nameNode}${workingDir}/coansys-citationmatching/output'. -->
-                    <value>${nameNode}${workingDir}/coansys/matching</value>
-                </property>
-            </configuration>
-        </sub-workflow>
-        <ok to="protobuf_to_avro_and_reference_attacher"/>
+            <jar>${oozieTopWfApplicationPath}/lib/iis-workflows-citationmatching-${projectVersion}.jar</jar>
+        
+            <arg>-citationPath = ${input}</arg>
+            <arg>-documentPath = ${input}</arg>
+            
+            <arg>-hashGeneratorClasses = pl.edu.icm.coansys.citations.hashers.CitationNameYearPagesHashGenerator:pl.edu.icm.coansys.citations.hashers.DocumentNameYearPagesHashGenerator</arg>
+            <arg>-hashGeneratorClasses = pl.edu.icm.coansys.citations.hashers.CitationNameYearPagesHashGenerator:pl.edu.icm.coansys.citations.hashers.DocumentNameYearNumNumHashGenerator</arg>
+            <arg>-hashGeneratorClasses = pl.edu.icm.coansys.citations.hashers.CitationNameYearHashGenerator:pl.edu.icm.coansys.citations.hashers.DocumentNameYearStrictHashGenerator</arg>
+            <arg>-hashGeneratorClasses = pl.edu.icm.coansys.citations.hashers.CitationNameYearHashGenerator:pl.edu.icm.coansys.citations.hashers.DocumentNameYearHashGenerator</arg>
+            
+            <arg>-outputDirPath = ${output}</arg>
+            <arg>-maxHashBucketSize = 10000</arg>
+            <arg>-numberOfPartitions = 5</arg>
+            
+            <arg>-inputDocumentReaderClass = eu.dnetlib.iis.workflows.citationmatching.DocumentMetadataInputReader</arg>
+            <arg>-inputDocumentConverterClass = eu.dnetlib.iis.workflows.citationmatching.DocumentMetadataInputConverter</arg>
+            
+            <arg>-inputCitationReaderClass = eu.dnetlib.iis.workflows.citationmatching.ReferenceMetadataInputReader</arg>
+            <arg>-inputCitationConverterClass = eu.dnetlib.iis.workflows.citationmatching.ReferenceMetadataInputConverter</arg>
+            
+            <arg>-outputConverterClass = eu.dnetlib.iis.workflows.citationmatching.CitationOutputConverter</arg>
+            <arg>-outputWriterClass = eu.dnetlib.iis.workflows.citationmatching.CitationOutputWriter</arg>
+            
```

Comment:
Well, actually, it should be an external job, because otherwise we won't be able to add transformers in the future.